lib/game_logic.rb

```diff
@@ -0,0 +1,96 @@
+require_relative '../lib/board.rb'
+
+class GameLogic < Board
+  def horizontal_check(player_sym)
+    i = 0
+    win = false
+    while i < 3
+      x = 0
+      result = []
+      while x < 3
+        result << @board[i][x]
+        x += 1
+      end
+      p result
+      if result.all?(player_sym)
+        win = true
+        break
+      end
+      i += 1
+    end
+    win
+  end
```

Comment:
There is a better way to check for a winner and you should implement it on your other methods (vertical and diagonal). **Tip:** What if you had an array of all the possible winning combinations and you just check if the current state of your board matches one of those??
Please take another look at these very important methods and work on them to make them more efficient.